server/routerlicious/packages/services-core/src/storageNameAllocator.ts

```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IStorageNameAllocator {
```

Comment:
just a thought: alternatively, _IStorageNameAllocator_ and _IStorageNameRetriever_ could have been combined into _IStorageNameManager_ :)

Reply:
It was in in one class, but split into two for: Allocator only used in document flow, retriever used much more on other flow. So tend to keep them loose so I don't have to expose/inject a lot of unnecessary dependencies to the retriever